src/main/io/gimbal_control.c

```diff
@@ -0,0 +1,326 @@
+                            0);
+    } else {
+        pitch += scaleRange(rcData[PITCH] - rxConfig()->midrc, 0, GIMBAL_RC_SET_MAX,
+                            0,
+                            GIMBAL_PITCH_MIN * gimbalTrackConfig()->gimbal_pitch_rc_high_gain / 100);
+    }
+    yaw   += scaleRange(rcData[YAW] - rxConfig()->midrc, GIMBAL_RC_SET_MIN, GIMBAL_RC_SET_MAX,
+                        GIMBAL_YAW_MIN * gimbalTrackConfig()->gimbal_yaw_rc_gain / 100,
+                        GIMBAL_YAW_MAX * gimbalTrackConfig()->gimbal_yaw_rc_gain / 100);
+
+    pitch += scaleRange(rcData[THROTTLE] - rxConfig()->midrc, GIMBAL_RC_SET_MIN, GIMBAL_RC_SET_MAX,
+                        GIMBAL_PITCH_MIN * gimbalTrackConfig()->gimbal_pitch_rc_thr_gain / 100,
+                        GIMBAL_PITCH_MAX * gimbalTrackConfig()->gimbal_pitch_rc_thr_gain / 100);
+
+    // Apply offsets
+    roll  += GIMBAL_ROLL_MAX * gimbalTrackConfig()->gimbal_roll_offset / 100;
+    pitch += GIMBAL_PITCH_MAX * gimbalTrackConfig()->gimbal_pitch_offset / 100;
+    yaw   += GIMBAL_YAW_MAX * gimbalTrackConfig()->gimbal_yaw_offset / 100;
+
+    DEBUG_SET(DEBUG_GIMBAL, 3, roll);
+    DEBUG_SET(DEBUG_GIMBAL, 4, pitch);
+    DEBUG_SET(DEBUG_GIMBAL, 5, yaw);
+
+    // Constrain to set limits
+    gimbalCmdOut.roll  = constrain(roll, GIMBAL_ROLL_MIN * gimbalTrackConfig()->gimbal_roll_limit / 100,
+                                         GIMBAL_ROLL_MAX * gimbalTrackConfig()->gimbal_roll_limit / 100);
+    gimbalCmdOut.pitch = constrain(pitch, GIMBAL_PITCH_MIN * gimbalTrackConfig()->gimbal_pitch_low_limit / 100,
+                                          GIMBAL_PITCH_MAX * gimbalTrackConfig()->gimbal_pitch_high_limit / 100);
+    gimbalCmdOut.yaw   = constrain(yaw, GIMBAL_YAW_MIN * gimbalTrackConfig()->gimbal_yaw_limit / 100,
+                                        GIMBAL_YAW_MAX * gimbalTrackConfig()->gimbal_yaw_limit / 100);
+
+    gimbalCmdOut.mode = gimbalTrackConfig()->gimbal_stabilisation;
+    gimbalCmdOut.sens = gimbalTrackConfig()->gimbal_sensitivity;
+
+    uint16_t crc = gimbalCrc((uint8_t *)&gimbalCmdOut, sizeof(gimbalCmdOut) - 2);
+
+    gimbalCmdOut.crc = crc;
+
+    return true;
+}
+
+// Gimbal updates should be sent at 100Hz or the gimbal will self center after approx. 2 seconds
+void gimbalUpdate(timeUs_t currentTimeUs)
+{
+    static enum {GIMBAL_OP1, GIMBAL_OP2, GIMBAL_CMD} gimbalParseState = GIMBAL_OP1;
+    static timeUs_t lastRxTimeUs = 0;
+
+    if (!gimbalSerialPort) {
+        setTaskEnabled(TASK_GIMBAL, false);
+        return;
+    }
+
+    // Read bytes from the VTX gimbal serial data stream
+
+    uint32_t bytes =  serialRxBytesWaiting(gimbalSerialPort);
+
+    if (bytes > 0) {
+        lastRxTimeUs = currentTimeUs;
+        while (bytes--) {
+            uint8_t inData = serialRead(gimbalSerialPort);
+
+            // If the packet is a gimbalCmd_t structure then parse, otherwise pass through
+            switch (gimbalParseState) {
```

Comment:
(this will detect false GIMBAL_CMD_S in data, with 1/64k probability - not a big problem. But it will fail spectacularly if 0x5AA5 is part of acknowledged packet that must be sent to gimbal)
You can reduce possibility of errors by checking CRC below.

Reply:
CRC check added.